diff --git a/drupal/sites/all/modules/contrib/revisioning/views/revisioning_handler_filter_revision_state.inc b/drupal/sites/all/modules/contrib/revisioning/views/revisioning_handler_filter_revision_state.inc
index a38e920..1e62a5d 100644
--- a/drupal/sites/all/modules/contrib/revisioning/views/revisioning_handler_filter_revision_state.inc
+++ b/drupal/sites/all/modules/contrib/revisioning/views/revisioning_handler_filter_revision_state.inc
@@ -2,7 +2,7 @@
 /**
  * @file
  * revisioning_handler_filter_revision_state.inc
- * 
+ *
  * Views filter override to filter on revision state, i.e. pending, archived or
  * current.
  */
@@ -45,13 +45,6 @@ class revisioning_handler_filter_revision_state extends views_handler_filter_in_
 
         case REVISION_PENDING:
           $subclauses[] = "($revisions_table.vid>$node_table.vid OR ($node_table.status=0 AND (SELECT COUNT(vid) FROM {" . $revisions_table . "} WHERE nid=$node_table.nid)=1))";
-          // Only add this join if there is not already a 'Revision NID of the
-          // content revision' contextual filter relationship.
-          if (empty($this->query->relationships[$this->relationship])) {
-            // Make sure UNIQUE is set!
-            $this->query->table_queue[$revisions_table]['join'] = new views_join();
-            $this->query->table_queue[$revisions_table]['join']->construct($node_table, $revisions_table, 'nid', 'nid');
-          }
           break;
       }
     }
